apps/namadillo/src/atoms/masp/functions.ts

```diff
@@ -0,0 +1,36 @@
+import { Asset } from "@chain-registry/types";
+import BigNumber from "bignumber.js";
+import { TokenBalance } from "./atoms";
+
+/**
+ * Find the expoent of a denom (`uatom`, `atom`, etc).
+ * Useful to calculate the token amount with correct decimal part
+ * @param asset
+ * @param denom
+ * @returns The expoent, or zero if not found
+ */
+export const findExpoent = (asset: Asset, denom: string): number =>
```

Comment:
Nitpick: Maybe these functions could be moved to integrations or other place more generic than MASP